### PR TITLE
Update AliCloud for VPC support and some small changes

### PR DESCRIPTION
**Pull request by hicrazyboy**
1. Support perf test in VPC network in AliCloud.
2. Fix a bug when creating instance with I/O optimized
3. Prefix the flag with ali_
https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/611/files#diff-f0e4cdc27d0409f2edd6208085ce97cbR43

### CODE REVIEW

File: perfkitbenchmarker/alicloud/ali_disk.py [hunk 1]
```diff
@@ -108,7 +111,7 @@ def Attach(self, vm):
         '--RegionId %s' % self.region,
         '--InstanceId %s' % self.attached_vm_id,
         '--DiskId %s' % self.id,
-        '--Device %s' % self.GetDevicePath()]
+        '--Device /dev/xvdb']
```
Comment (voellm) [inline perfkitbenchmarker/alicloud/ali_disk.py]: Did you mean to hardcode this path?

File: perfkitbenchmarker/alicloud/ali_disk.py [hunk 2]
```diff
@@ -131,4 +134,27 @@ def Detach(self):
 
   def GetDevicePath(self):
     """Returns the path to the device inside the VM."""
-    return '/dev/xvd%s' % self.device_letter
+    if FLAGS.ali_io_optimized == 'none':
```
Comment (voellm) [inline perfkitbenchmarker/alicloud/ali_disk.py]: If you change https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/alicloud/ali_virtual_machine.py#L43 from 'none' to be None then you can write the code list this:
if FLAGS.ali_io_optimized is not None:]
...

or if you dont list the negative switch the then and else causes for
if FLAGS.ali_io_optimized:
  return '/dev/vd%s' % self.device_letter
else:
  return '/dev/xvd%s' % self.device_letter

File: perfkitbenchmarker/alicloud/ali_disk.py [hunk 3]
```diff
@@ -131,4 +134,27 @@ def Detach(self):
...
+      return '/dev/xvd%s' % self.device_letter
+    else:
+      return '/dev/vd%s' % self.device_letter
+
+  @vm_util.Retry(poll_interval=1, log_errors=False)
+  def WartForDiskStatus(self, status_list):
```
Comment (voellm) [inline perfkitbenchmarker/alicloud/ali_disk.py]: Wart or Wait ?

File: perfkitbenchmarker/alicloud/ali_disk.py [hunk 4]
```diff
@@ -131,4 +134,27 @@ def Detach(self):
...
+    """Waits until disk is attach to the instance"""
+    logging.info('Waits until the disk\'s stastus is one of statuses: %s',
+                 status_list)
+    attach_cmd = util.ALI_PREFIX + [
+        'ecs',
+        'DescribeDisks',
+        '--RegionId %s' % self.region,
+        '--ZoneId %s' % self.zone,
+        '--DiskIds \'["%s"]\'' % self.id,
+        '--Device /dev/xvdb']
```
Comment (voellm) [inline perfkitbenchmarker/alicloud/ali_disk.py]: Do you really want this hardcoded?

File: perfkitbenchmarker/alicloud/util.py [hunk 5]
```diff
@@ -93,7 +93,7 @@ def AddDefaultTags(resource_id, resource_type, region):
   AddTags(resource_id, resource_type, region, **tags)
 
 
-@vm_util.Retry()
+@vm_util.Retry(log_errors=False)
```
Comment (voellm) [inline perfkitbenchmarker/alicloud/util.py]: Why false here?

**Comment by voellm** [comment]
@ehankland  Can you review this as well.  Thanks.

File: perfkitbenchmarker/alicloud/util.py (hunk 5 above)
Comment (hicrazyboy) [inline perfkitbenchmarker/alicloud/util.py]: Ok, true may be better.

File: perfkitbenchmarker/alicloud/ali_disk.py (hunk 3 above)
Comment (hicrazyboy) [inline perfkitbenchmarker/alicloud/ali_disk.py]: A mistake.

File: perfkitbenchmarker/alicloud/ali_disk.py (hunk 4 above)
Comment (hicrazyboy) [inline perfkitbenchmarker/alicloud/ali_disk.py]: I will replace it with function.

**Comment by hicrazyboy** [comment]
Thank you for review again, I just updated.

File: perfkitbenchmarker/alicloud/ali_network.py [hunk 6]
```diff
@@ -29,18 +30,139 @@
...
 
+flags.DEFINE_boolean('ali_use_vpc', True,
+                     'Use VPC to create networks')
 
 FLAGS = flags.FLAGS
 MAX_NAME_LENGTH = 128
 
 
+class AliVpc(resource.BaseResource):
+  """An object representing an Aws VPC."""
```
Comment (ehankland) [inline perfkitbenchmarker/alicloud/ali_network.py]: nit: update docstring

File: perfkitbenchmarker/alicloud/ali_network.py [hunk 7]
```diff
@@ -29,18 +30,139 @@
...
+    describe_cmd = util.GetEncodedCmd(describe_cmd)
+    stdout, _ = vm_util.IssueRetryableCommand(describe_cmd)
+    response = json.loads(stdout)
+    vswitches = response['VSwitches']['VSwitch']
+    assert len(vswitches) < 2, 'Too many VSwitches.'
+    return len(vswitches) > 0
+
+
 class AliSecurityGroup(resource.BaseResource):
   """Object representing an Azure Affinity Group."""
```
Comment (ehankland) [inline perfkitbenchmarker/alicloud/ali_network.py]: nit: update docstring

File: perfkitbenchmarker/alicloud/ali_virtual_machine.py [hunk 8]
```diff
@@ -152,10 +234,19 @@ def _PostCreate(self):
...
+      self.internal_ip = \
+          instance['VpcAttributes']['PrivateIpAddress']['IpAddress'][0]
+    else:
+      pub_ip_address = instance['PublicIpAddress']['IpAddress'][0]
+      self.internal_ip = instance['InnerIpAddress']['IpAddress'][0]
+    assert self.ip_address == pub_ip_address
     self.group_id = instance['SecurityGroupIds']['SecurityGroupId'][0]
 
+    self._WaitForInstanceStatus(['Running'])
+    self.bootable_time = time.time()
```
Comment (ehankland) [inline perfkitbenchmarker/alicloud/ali_virtual_machine.py]: This should only be set after we've successfully sshed into the VM - it should get set for you at the appropriate time, so you don't need this line

File: perfkitbenchmarker/alicloud/ali_network.py [hunk 9]
```diff
@@ -85,41 +210,13 @@ def _Exists(self):
...
-          '--IpProtocol %s' % proto,
-          '--PortRange %s/%s' % (port, port),
-          '--SourceCidrIp 0.0.0.0/0',
-          '--RegionId %s' % self.region,
-          '--SecurityGroupId %s' % self.group_id]
-      allow_cmd = util.GetEncodedCmd(allow_cmd)
-      vm_util.IssueRetryableCommand(allow_cmd)
 
 
 class AliFirewall(network.BaseFirewall):
```
Comment (ehankland) [inline perfkitbenchmarker/alicloud/ali_network.py]: Given the fact that you are now opening up all ports when you create the VM, I don't think you need this anymore

File: perfkitbenchmarker/alicloud/ali_virtual_machine.py (hunk 8 above)
Comment (hicrazyboy) [inline perfkitbenchmarker/alicloud/ali_virtual_machine.py]: all right

**Comment by hicrazyboy** [comment]
@ehankland @voellm 
Can you review the last commit  'b1e6ea5' ? thanks!

File: perfkitbenchmarker/providers/alicloud/ali_virtual_machine.py [hunk 10]
```diff
@@ -142,10 +214,19 @@ def _PostCreate(self):
...
+          instance['VpcAttributes']['PrivateIpAddress']['IpAddress'][0]
+    else:
+      pub_ip_address = instance['PublicIpAddress']['IpAddress'][0]
+      self.internal_ip = instance['InnerIpAddress']['IpAddress'][0]
+    assert self.ip_address == pub_ip_address
     self.group_id = instance['SecurityGroupIds']['SecurityGroupId'][0]
 
+    self._WaitForInstanceStatus(['Running'])
+
+    self.firewall.AllowPort(self, SSH_PORT)
```
Comment (ehankland) [inline perfkitbenchmarker/providers/alicloud/ali_virtual_machine.py]: Can you confirm whether this is necessary? Port 22 should get opened here: https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/benchmark_spec.py#L288

Comment (ehankland) [inline perfkitbenchmarker/providers/alicloud/ali_virtual_machine.py]: Ah - is this because of util.AddPubKeyToHost?

**Comment by ehankland** [comment]
LGTM - I recommend squashing some of the commits to make the history cleaner. Let me know when you want me to merge

File: perfkitbenchmarker/providers/alicloud/ali_virtual_machine.py (hunk 10 above)
Comment (hicrazyboy) [inline perfkitbenchmarker/providers/alicloud/ali_virtual_machine.py]: Yeah

**Comment by hicrazyboy** [comment]
How about today?  What can I do for you?
Thanks!

**Comment by hicrazyboy** [comment]
correct some mistakes in README.md

**Comment by ehankland** [comment]
Merged - thanks!

**Pull request merged**